Create SECURITY.md (#366)
drand/drand: SECURITY.md

# Security Policy

## Supported Versions

> The Drand team will be actively supporting the most recent minor version of drand.

Reviewer: ```suggestion
The Drand team actively supports the most recent minor version of drand.
```

> Upcoming major or minor releases are announced with a prerelease at least one week in advance, and are communicated through the leagueofentropy@googlegroups.com email list.

Reviewer 2: Do we really have to tie our hands by saying this publicly ? If there's an urgent patch, I wouldn't want to wait one week. Usually, ppl release new versions and then tell the world and there can be a migration plan etc.
Im not sure I see the necessity of a warning _in advance_

Author: once we have clients using the client library, we need to let them know in advance if a release fixes a security issue, especially if protocol breaking. I think that's the main point here.

Reviewer 3: @willscott is right here. Doing proper release management is part of making a software library production ready.

See the level of detail that go-ipfs goes into about their release process at -- https://github.com/ipfs/go-ipfs/blob/master/docs/releases.md --.

We should not tie the release process into a specific mailing list of adopters, but craft a release process that is public (e.g. Github issue) and then treat the LoE as one of the adopters (equivalent to stage 3 from https://github.com/ipfs/go-ipfs/blob/master/docs/releases.md#stage-3---community-prod-testing).

Reviewer 2: Ok sounds good - and yes for LoE -we should post it on drand.love or stg

Author: where are we going to communicate this? if loe email list pointed here isn't publicly joinable, it doesn't seem like a great announcement channel.

> Upcoming major or minor releases will be announced with a prerelease at least one week in advance, and will be communicated through the leagueofentropy@googlegroups.com email list.

Reviewer: ```suggestion
Upcoming major or minor releases are announced with a prerelease at least one week in advance, and are communicated through the leagueofentropy@googlegroups.com email list.
```

> Upcoming major or minor releases will be announced with a prerelease at least one week in advance, and will be communicated through the leagueofentropy@googlegroups.com email list.

Reviewer: Add link to sign up?

Author: https://groups.google.com/forum/#!forum/leagueofentropy appears to be visible archives, but not open signup. @daviddias  is there a better place for people to be notified? maybe just subscribe to notifications of releases on the github repo? we can't do that in advance though.

Reviewer 3: I believe we shouldn't rely on the LoE mlist as the primary channel. See my comment at https://github.com/drand/drand/pull/366#discussion_r429782575

## Reporting a Vulnerability

> To report a security issue, please file an issue in this repository, and indicate that it is a vulnerability or security issue.

Reviewer 2: I would simply say to directly send to the email . A  person shouldn't be the judge whether it's too serious of an issue or not to be disclosed publicly; by default it shouldn't be disclosed publicly, and it's up to us to judge whether that's a serious or not issue.

If you believe the issue is sensitive or can cause biasability, denial of service, or compromise of the active network,

> please do not directly provide details, but instead provide a method for contacting you off-github, and the drand

Reviewer 3: What about inviting them to send an email to drand@protocol.ai?

Reviewer 2: Yes

Reviewer 3: will get a sec mailing list next week

contributors will reach out to get details from you and work towards a resolution.

> We commit to updating security issues filed on github in this way on a weekly basis to indicate their current status towards resolution.

Reviewer 2: I would try to commit as few things as possible - this is also a commitment I dont see any reason to exist: 
from what i see in other oss projects, one guy report an issue privately, we fix it, then we contact operators to upgrade, then we announce the issue + resolution after a while ?

Reviewer 2: Announcing we are working towards "this" security issue just raises a red flag that there is indeed a security issue here to be attacked.

Author: i think it is somewhat up to the external person who's found the issue. There are plenty of sec people who expect money if they report privately / expect that to imply the existence of a bug bounty program.

It's worth having some sort of "we're willing to come to you if you want to be more responsible than dropping a full-disclosure", and are happy to be transparent in fixing reported issues.

I can take another pass at wording, if you have suggested wording, also happy to take that :)

Reviewer 3: I've updated the wording to be about sending an email, rather than opening an issue.

@nikkolasg the sad reality is that "standard way things get done in Open Source" isn't that great or sophisticated and that is why there are so many issues unresolved that are actually massive exploits for someone not putting the time. Running a thing that we say production ready means that we need to have a platform for a Responsible Disclosure Program.

Reviewer 2: @daviddias many open sources programs have a responsible disclosure program, [ zcash included ](https://github.com/zcash/zcash/blob/master/responsible_disclosure.md) :
> resolve those issues on an appropriate timeline and perform a coordinated release

They do not expose any security issue publicly before it has been resolved and announced to "big" upstream users. Every other project is doing something similar btw.

Here it seems to imply we are going to talk about security issues publicly to say how far are we with resolution ? I hope not, we should not do this. And if it not the case, then I would like to re-word this sentence to better explain what we intend to do.

> We will commit to updating security issues filed on github in this way on at least a weekly basis to indicate their current status towards resolution.

Reviewer: ```suggestion
We commit to updating security issues filed on github in this way on a weekly basis to indicate their current status towards resolution.
```

# Security Policy

## Supported Versions

The Drand team actively supports the most recent minor version of drand.

Upcoming major or minor releases are announced with a prerelease at least one week in advance, and are communicated through the leagueofentropy@googlegroups.com email list.

## Reporting a Vulnerability

If you believe to have found a security issue that can cause biasability, denial of service, or compromise of the active network, please send us an email to security@drand.love. We'll make sure to get back to you ASAP, work towards verifying the issue, mitigating it and providing an update for our users. We commit to updating security issues filed on github in this way on a weekly basis to indicate their current status towards resolution.

To help us serving the drand ecosystem, we appreciate that your reports are as precise as possible, ideally containing a way to reproduce the vulnerability.

> For all other kinds of bugs, please fill an issue in this repo. 

Reviewer: ```suggestion
For all other kinds of bugs, please file an issue in this repo. 
```

# Security Policy

## Supported Versions

The Drand team actively supports the most recent minor version of drand.

> Upcoming major or minor releases are announced with a prerelease at least one week in advance, and are communicated through the leagueofentropy@googlegroups.com email list.

Reviewer 3: ```suggestion
Upcoming major or minor releases will go through a release-candidate cycle before they are released to the ecosystem.```
